tests/bricks/test_sequence_generators.py

```diff
@@ -101,18 +120,35 @@ def test_integer_sequence_generator():
         seed=1234)
     generator.initialize()
 
-    # Test cost
+    # Test 'cost_matrix' method
     y = tensor.lmatrix('y')
     mask = tensor.matrix('mask')
-    costs = generator.cost(y, mask)
+    costs = generator.cost_matrix(y, mask)
     assert costs.ndim == 2
     costs_fun = theano.function([y, mask], [costs])
-    costs_val = costs_fun(
-        rng.randint(readout_dim, size=(n_steps, batch_size)),
-        numpy.ones((n_steps, batch_size), dtype=floatX))[0]
+    y_test = rng.randint(readout_dim, size=(n_steps, batch_size))
+    m_test = numpy.ones((n_steps, batch_size), dtype=floatX)
+    costs_val = costs_fun(y_test, m_test)[0]
     assert costs_val.shape == (n_steps, batch_size)
     assert_allclose(costs_val.sum(), 482.827, rtol=1e-5)
 
+    # Test 'cost' method
+    cost = generator.cost(y, mask)
+    assert cost.ndim == 0
+    cost_val = theano.function([y, mask], [cost])(y_test, m_test)
+    assert_allclose(cost_val, 16.0942, rtol=1e-5)
+
+    # Test 'AUXILIARY' variable 'per_sequence_element' in 'cost' method
+    cg = ComputationGraph([cost])
+    var_filter = VariableFilter(roles=[AUXILIARY])
+    aux_var_name = '_'.join([generator.name, generator.cost.name,
+                             'per_sequence_element'])
+    cost_per_el = [el for el in var_filter(cg.variables)
```

Comment:
We can leave it, but just a hint: there is a class `VariableFilter` that can help your to find a variable with a given `x.tag.name`, i.e. `per_sequence_element` in this case.